### DISCUSSION TITLE
setting publicPath breaks devServer

### DISCUSSION THREAD

**Discussion by charlycou** [quasarframework/quasar#15838] [General - Components / Directives / etc] [answered]
I have a simple SPA that needs to be deployed in a nested directory. I'm setting `build.publicPath` property in quasar.config.js and it works perfectly fine when the SPA is deployed. 

```
build: {
	publicPath: '/association-ui/'
	vueRouterMode: 'history'
}
devServer: {
	proxy: {
		'/association': 'http://localhost:8080'
	}
}
```
However, it breaks the app locally while running on devServer. `publicPath` seems to affect `vueRouterBase` since http://localhost is redirected to http://localhost/association-ui/ which answer 404 not found. When `publicPath` is disabled the application can be reached at http://localhost as expected.
**How should I configure devServer to take into account `publicPath` property? I would like to run the same configuration locally and when building the app.**

**Comment by suleimanas** [comment, marked as answer]
one way is to set your public path on production only, something like:

```
module.exports = configure(function (ctx) {
    return {
        build: {
          publicPath: ctx.prod ? '/association-ui/':'/'
        }
    };
});
```

edit: sorry i missed the last part of your question
https://quasar.dev/quasar-cli-vite/api-proxying
https://quasar.dev/quasar-cli-webpack/api-proxying

**Reply by charlycou** [reply]
Exactly what I was looking for. Sorry for the noise.

**Comment by MusulmonLolayev** [comment]
This only works partially that the links are not properly updated for other files in folder assets, including *.js, *.icons, etc. How can I fully achieve this? Thank you in advance.
